src/Chainweb/SPV/CreateProof.hs

```diff
@@ -484,7 +488,13 @@ minimumTrgHeader headerDb tcid scid bh = do
             }
   where
     trgChain = headerDb ^?! ixg tcid
-    trgHeight = int bh + int (length path)
+    trgHeight
+        | srcGraph == trgGraph = int bh + int distance
+        | otherwise = int bh + 2 * (int distance)
+            -- This assumes that graph changes are at least graph-diameter
+            -- blocks appart.
+
```

Comment:
This changes guarantees that an SPV proofs can be constructed on histories with graph changes.